plenum/common/stacks.py

```diff
@@ -38,18 +39,35 @@ def __init__(self, stackParams: dict, msgHandler: Callable, seed=None,
             onlyListener=True,
             config=config,
             msgRejectHandler=msgRejectHandler,
-            create_listener_monitor=True)
+            create_listener_monitor=config.TRACK_CONNECTED_CLIENTS_NUM_ENABLED)
         MessageProcessor.__init__(self, allowDictOnly=False)
 
+        if config.CLIENT_STACK_RESTART_ENABLED and not config.TRACK_CONNECTED_CLIENTS_NUM_ENABLED:
+            error_str = '{}: client stack restart is enabled (CLIENT_STACK_RESTART_ENABLED) ' \
+                        'but connections tracking is disabled (TRACK_CONNECTED_CLIENTS_NUM_ENABLED), ' \
+                        'please check your configuration'.format(self)
+            raise RuntimeError(error_str)
+
+        self.track_connected_clients_num_enabled = config.TRACK_CONNECTED_CLIENTS_NUM_ENABLED
```

Comment:
Should it be `self.track_connected_clients_num_enabled=create_listener_monitor`?

Reply:
The create_listener_monitor is an input parameter of ZStack class while config.TRACK_CONNECTED_CLIENTS_NUM_ENABLED is a configuration parameter, I don't understand why it should be replaced.